java/core/src/main/java/sleeper/core/range/Range.java

```diff
@@ -194,8 +194,36 @@ public int hashCode() {
         return hash;
     }
 
+    /**
+     * Determines if the contents of the Range are matches.
+     * For checking if objects are exact matches please use equalsExactMatch.
+     *
+     * @param  obj comparison object
+     * @return     if contents are equal
+     */
     @Override
     public boolean equals(Object obj) {
+        if (equalsExactMatch(obj)) {
+            return true;
+        } else {
+            return equalsCanonicalised(this, obj);
+        }
```

Comment:
It seems a bit confusing to do both checks here? I think it'd be clearer if we just have the canonicalised implementation here.